pkg/api/resource_helpers_test.go

```diff
@@ -118,3 +118,111 @@ func TestIsPodAvailable(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdatePodCondition(t *testing.T) {
+	tests := []struct {
+		podStatus    *PodStatus
+		podCondition *PodCondition
+		expected     bool
+	}{
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodInitialized,
+						Status: ConditionFalse,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: true,
+		},
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodReady,
+						Status: ConditionFalse,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: true,
+		},
+		{
+			podStatus: &PodStatus{
+				Conditions: []PodCondition{
+					{
+						Type:   PodReady,
+						Status: ConditionTrue,
+					},
+				},
+			},
+			podCondition: &PodCondition{
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: false,
+		},
+	}
+
+	for i, test := range tests {
+		changedOrAadd := UpdatePodCondition(test.podStatus, test.podCondition)
+		if changedOrAadd != test.expected {
```

Comment:
`PodCondition.LastTransitionTime`  in  the function https://github.com/kubernetes/kubernetes/blob/master/pkg/api/resource_helpers.go#L147-L147 has been set current  local time . When pod condition has changed or has been added, this field can not be equal.
How to solve it making the actual value and expected value DeepEqual ?
Thanks.